src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -46,38 +46,25 @@ function useDecline () {
   })
 }
 
-function useFetchCounterparties () {
+function useTransactionsWithCounterparties () {
   const { data: { holofuelUser: whoami = {} } = {} } = useQuery(HolofuelUserQuery)
+  const { data: { holofuelInboxCounterparties = [] } = {} } = useQuery(HolofuelInboxCounterpartiesQuery)
   const { data: { holofuelActionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
   const { data: { holofuelNonPendingTransactions = [] } = {} } = useQuery(HolofuelNonPendingTransactionsQuery)
-  const { data: { holofuelInboxCounterparties } = {}, client } = useQuery(HolofuelInboxCounterpartiesQuery)
-
-  if (holofuelInboxCounterparties) {
-    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
-    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
-      const matchingTx = filterTransactionsByAgentId(agent, txListType)
-      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
-    })
-
-    // Cache Write/Update for holofuelActionableTransactions
-    const reflexiveActionableTxList = updateTxListCounterparties(holofuelActionableTransactions, [whoami])
-    const newActionableTxList = flatten((updateTxListCounterparties(holofuelActionableTransactions, holofuelInboxCounterparties)).concat(reflexiveActionableTxList))
```

Comment:
this line was the cause of the bug.

If you have sent a transaction to yourself, both `holofuelActionableTransactions` and `reflexiveActionableTxList` have a copy of the transaction. So then when you write the new list to the cache, it's double in length, and so this query reruns, and now they each have 2 copies of the same transaction in them and so `newActionableTxList` has four copies, and the whole process feeds back on itself, with the transaction list doubling in size every time.

The fix for this is a little hidden in the updated code because I changed a lot of how this function works, but the `uniqBy` in line 60 prevents the same thing happening in the new code.

The very simplest version of the bug fix would be to add a `uniqBy` to line 64 in the old code.

Reply:
Good catch!  This was definitely the culprit! :tada: